app/main.py

```diff
@@ -2,0 +1,42 @@
+from types import TracebackType
+
+
+class Animal:
+    alive = []
+
+    def __init__(self, name: str, health: int = 100) -> None:
+        self.name: str = name
+        self.health: int = health
+        self.hidden: bool = False
+        Animal.alive.append(self)
+
+    def __enter__(self) -> None:
+        pass
+
+    def __exit__(
+            self,
+            exc_type: type,
+            exc_val: Exception,
+            exc_tb: TracebackType
+    ) -> None:
+        if self in Animal.alive:
+            Animal.alive.remove(self)
+
+    def __repr__(self) -> str:
```

Comment:
According to the checklist, use the class name to get access to the class attributes. So, instead of `self.alive.append(self)`, you should use `Animal.alive.append(self)`. This is a code efficiency issue.